File: work/abc084_d.py
# -*- coding: utf-8 -*-
Q = int(input())
query = []
rmax = -1
for _ in range(Q):
    line = list(map(int, input().split()))
    query.append(line)
    rmax = max(rmax, line[1])

prime = [2]

for x in range(3, rmax+1):
    pflag = True
    for n in prime:
        if x%n==0:
            pflag = False
            break
        if x<n*n:
            break
    if pflag:
        prime.append(x)
# print(prime)
acc = [0 for _ in range(rmax+1)]
for n in range(rmax+1):
    if n in prime and (n+1)//2 in prime:
        acc[n] += 1
    if n<rmax:
        acc[n+1] += acc[n]
# print(acc)
for q in query:
    print(acc[q[1]] - acc[q[0]-1])

File: work/abc084_d.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main(){
  int Q;
  cin>>Q;
  int rmax;
  vector<int> qr(Q);
  vector<int> ql(Q);
  for(int i=0;i<Q;i++){
    cin>>ql[i]>>qr[i];
    if(qr[i]>rmax){
      rmax = qr[i];
    }
  }

  vector<int> prime;
  prime.push_back(2);
  for(int x=3;x<rmax+1;x++){
    bool pflag = true;
    for(auto n: prime){
      if(x%n==0){
        pflag = false;
        break;
      }
      if(x<n*n){
        break;
      }
    }
    if(pflag){
      prime.push_back(x);
    }
  }

  vector<int> acc(rmax+1, 0);
  for(int n=0;n<rmax+1;n++){
    if(find(prime.begin(),prime.end(),n)!=prime.end()
      && find(prime.begin(),prime.end(), (n+1)/2)!=prime.end()){
      acc[n] += 1;
    }
    if(n<rmax){
      acc[n+1] += acc[n];
    }
  }

  // for(auto x: prime){
  //   cout<<x<<" ";
  // }
  // cout<<endl;
  // for(auto x: acc){
  //   cout<<x<<" ";
  // }
  // cout<<endl;
  for(int i=0;i<Q;i++){
    cout<<acc[qr[i]]-acc[ql[i]-1]<<endl;
  }
}
